pkgs/development/haskell-modules/lib/compose.nix

```diff
@@ -463,4 +463,59 @@ rec {
   allowInconsistentDependencies = overrideCabal (drv: {
     allowInconsistentDependencies = true;
   });
+
+  # The motivation for this utility is for use for CI builds in order to avoid a
+  # full rebuild on every commit to the trunk development branch or every pull
+  # request.  For more details, see:
+  #
+  # https://harry.garrood.me/blog/easy-incremental-haskell-ci-builds-with-ghc-9.4/
+  #
+  # This accelerates a Haskell package build by building the package
+  # "incrementally", meaning that a "full" rebuild is only done once every
+  # interval and all rebuilds in between are "incremental", meaning that each
+  # incremental build reuses the `dist` directory from the last full rebuild.
+  #
+  # This only works for packages that use `git` for their source.
+  incremental = { interval }: pkg:
+    let
+      requiredNixVersion = "2.12.0pre20221128_32c182b";
```

Comment:
Please remove this function from the PR. If I understand it correctly, it depends on https://github.com/NixOS/nix/pull/7362 – “presuming it is merged” is presumed too much. In general I also want to keep anything out of `haskell.lib` that depends on unreleased Nix features.

Presumed too much also in the sense that I believe the `fetchGit` feature will be quite controversial.